src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/AsnEncodedDataEnumerator.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    public sealed class AsnEncodedDataEnumerator : IEnumerator
+    {
+        internal AsnEncodedDataEnumerator(AsnEncodedDataCollection asnEncodedDatas)
+        {
+            _asnEncodedDatas = asnEncodedDatas;
+            _current = -1;
+        }
+
+        public AsnEncodedData Current
+        {
+            get
+            {
+                return _asnEncodedDatas[_current];
+            }
+        }
+
+        object IEnumerator.Current
+        {
+            get
+            {
+                return _asnEncodedDatas[_current];
+            }
+        }
+
+        public bool MoveNext()
+        {
+            if (_current == _asnEncodedDatas.Count - 1)
```

Comment:
This is a faithful port, but `>=` is safer (particularly since this enumerator doesn't otherwise guard against modification)